crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -986,9 +986,17 @@ fn generate_tangents_for_mesh(mesh: &Mesh) -> Result<Vec<[f32; 4]>, GenerateTang
             ))
         }
     };
-    let indices = mesh
-        .indices()
-        .ok_or(GenerateTangentsError::MissingIndices)?;
+
+    let indices = if mesh.indices().is_none() {
+        if positions.len() < u16::MAX as usize {
+            Indices::U16((0..positions.len()).map(|i| i as u16).collect())
+        } else {
+            Indices::U32((0..positions.len()).map(|i| i as u32).collect())
+        }
+    } else {
+        Indices::U16(Default::default())
+    };
+    let indices = mesh.indices().unwrap_or(&indices);
```

Comment:
how about something like: 
```suggestion
    let indices = mesh.indices().unwrap_or_else(|| {
      	if positions.len() < u16::MAX as usize {
            Indices::U16((0..positions.len()).map(|i| i as u16).collect())
        } else {
            Indices::U32((0..positions.len()).map(|i| i as u32).collect())
        }
    });
```

That way it only runs the code when there's not indices

Reply:
Does that run? I had an error about returning a reference to a temporary when I tried

Reply:
Oh, I haven't tried it directly, I was just suggesting trying something along those line

Reply:
yeah, it doesn't run. 

i'd note that Vec::default() doesn't allocate, so there's only a bit of stack being used when the indices already exist.

Reply:
```suggestion
   let indices = if let Some(indices) = mesh.indices() {
        indices
    } else {
        if positions.len() < u16::MAX as usize {
            Indices::U16((0..positions.len()).map(|i| i as u16).collect())
        } else {
            Indices::U32((0..positions.len()).map(|i| i as u32).collect())
        }
    };
```

How about something like this then? I'm not sure if you can return the indices like that though.

Anyway, doesn't really matter, I was just curious if it was possible to condense it a bit and maybe remove that default branch that seem unnecessary.

Reply:
Dang, alright, I approved either way, but would have been nice to avoid that. Doesn't really matter though.

Reply:
pushing the option through to the helper avoids this and seems cleaner, thanks for pushing me :)

Reply:
oh nice, yeah, it seems like a good solution